File: src/main/java/org/digma/intellij/plugin/jaegerui/model/GoToSpanMessage.java
package org.digma.intellij.plugin.jaegerui.model;

import com.fasterxml.jackson.annotation.*;
import com.fasterxml.jackson.databind.annotation.JsonSerialize;


@JsonIgnoreProperties(ignoreUnknown = true)
@JsonInclude(JsonInclude.Include.NON_NULL)
@JsonAutoDetect(fieldVisibility = JsonAutoDetect.Visibility.ANY)
@JsonSerialize
public record GoToSpanMessage(@JsonProperty(value = "action",required = true) String action,
                              @JsonProperty(value = "payload",required = true) Span payload) {

    @JsonCreator(mode = JsonCreator.Mode.PROPERTIES)
    public GoToSpanMessage {
        //nothing to do , for jackson only
    }
}

File: src/main/kotlin/org/digma/intellij/plugin/ui/jaegerui/model/GoToSpanMessage.kt
package org.digma.intellij.plugin.ui.jaegerui.model

import com.fasterxml.jackson.annotation.JsonCreator
import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import java.beans.ConstructorProperties

@JsonIgnoreProperties(ignoreUnknown = true)
data class GoToSpanMessage
@JsonCreator(mode = JsonCreator.Mode.PROPERTIES)
@ConstructorProperties("action", "payload")
constructor(
    val action: String,
    val payload: Span
)

